ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitter.scala

```diff
@@ -614,4 +632,17 @@ private[kvutils] object TransactionCommitter {
   def getContractState(commitContext: CommitContext, key: DamlStateKey): DamlContractState =
     commitContext.get(key).getOrElse(throw Err.MissingInputState(key)).getContractState
 
+  @SuppressWarnings(Array("org.wartremover.warts.Option2Iterable"))
+  private def transactionMinimumRecordTime(
+      maybeDeduplicateUntil: Option[Instant],
+      ledgerTime: Instant,
+      maxSkew: Duration,
+      submissionTime: Instant): Instant =
+    List(
+      maybeDeduplicateUntil,
```

Comment:
I think there's an off-by-one error here. Min record time and max record time are inclusive. However, the deduplication time window is inclusive on the "reject side"; at least that's what @miklos-da implemented in https://github.com/digital-asset/daml/pull/6568/files#diff-51a916eda78c1c741b9b676658bb9132R277 and in Sandbox https://github.com/digital-asset/daml/blob/master/ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/inmemory/InMemoryLedger.scala#L580.

So you want to use `maybeDeduplicateUntil.map(_ + leastMeasurableTimeUnit)` here. AFAIK our time unit is microseconds.

Reply:
Thanks, addressed.